tests/cpptests/test_cpp_iterator_union.cpp

```diff
@@ -0,0 +1,264 @@
+        ASSERT_EQ(ui->base.current->docId, resultSet[i]);
+        ASSERT_EQ(ui->base.LastDocId, resultSet[i]);
+        ASSERT_TRUE(ui->base.isValid);
+        i++;
+    }
+    ASSERT_EQ(rc, ITERATOR_EOF);
+    ASSERT_FALSE(ui->base.isValid);
+    ASSERT_EQ(ui_base->Read(ui_base), ITERATOR_EOF); // Reading after EOF should return EOF
+    ASSERT_EQ(i, resultSet.size()) << "Expected to read " << resultSet.size() << " documents";
+
+    size_t expected = 0;
+    for (auto &child : docIds) {
+        expected += child.size();
+    }
+    ASSERT_EQ(ui->nExpected, expected);
+    ASSERT_EQ(ui_base->NumEstimated(ui_base), expected);
+}
+
+TEST_P(UnionIteratorCommonTest, SkipTo) {
+    UnionIterator *ui = (UnionIterator *)ui_base;
+    IteratorStatus rc;
+    // Test skipping to any id between 1 and the last id
+    t_docId i = 1;
+    for (t_docId id : resultSet) {
+        while (i < id) {
+            ui_base->Rewind(ui_base);
+            rc = ui_base->SkipTo(ui_base, i);
+            ASSERT_EQ(rc, ITERATOR_NOTFOUND);
+            ASSERT_EQ(ui->base.LastDocId, id);
+            ASSERT_EQ(ui->base.current->docId, id);
+            i++;
+        }
+        ui_base->Rewind(ui_base);
+        rc = ui_base->SkipTo(ui_base, id);
+        ASSERT_EQ(rc, ITERATOR_OK);
+        ASSERT_EQ(ui->base.LastDocId, id);
+        ASSERT_EQ(ui->base.current->docId, id);
+        i++;
+    }
+    // Test reading after skipping to the last id
+    ASSERT_EQ(ui_base->Read(ui_base), ITERATOR_EOF);
+    ASSERT_EQ(ui_base->SkipTo(ui_base, ui_base->LastDocId + 1), ITERATOR_EOF);
+    ASSERT_FALSE(ui->base.isValid);
+
+    ui_base->Rewind(ui_base);
+    ASSERT_EQ(ui->base.LastDocId, 0);
+    ASSERT_TRUE(ui->base.isValid);
+    // Test skipping to all ids that exist
+    for (t_docId id : resultSet) {
+        rc = ui_base->SkipTo(ui_base, id);
+        ASSERT_EQ(rc, ITERATOR_OK);
+        ASSERT_EQ(ui->base.LastDocId, id);
+        ASSERT_EQ(ui->base.current->docId, id);
+    }
+
+    // Test skipping to an id that exceeds the last id
+    ui_base->Rewind(ui_base);
+    ASSERT_EQ(ui->base.LastDocId, 0);
+    ASSERT_TRUE(ui->base.isValid);
+    rc = ui_base->SkipTo(ui_base, resultSet.back() + 1);
+    ASSERT_EQ(rc, ITERATOR_EOF);
+    ASSERT_EQ(ui->base.LastDocId, 0); // we just rewound
+    ASSERT_FALSE(ui->base.isValid);
+}
+
+TEST_P(UnionIteratorCommonTest, Rewind) {
+    UnionIterator *ui = (UnionIterator *)ui_base;
+    IteratorStatus rc;
+    for (int i = 0; i < 5; i++) {
+        for (int j = 0; j <= i; j++) {
+            ASSERT_EQ(ui_base->Read(ui_base), ITERATOR_OK);
+            ASSERT_EQ(ui->base.current->docId, resultSet[j]);
+            ASSERT_EQ(ui->base.LastDocId, resultSet[j]);
+        }
+        ui_base->Rewind(ui_base);
+        ASSERT_EQ(ui->base.LastDocId, 0);
+        ASSERT_TRUE(ui->base.isValid);
+    }
+}
+
+INSTANTIATE_TEST_SUITE_P(UnionIteratorP, UnionIteratorCommonTest, ::testing::Combine(
+    ::testing::Values(2, 5, 25),
```

Comment:
Can we validate somehow that we are running in "heap mode" when we expect to (so if configuration changes somehow we still have the coverage)